### ISSUE TITLE
Spring security tomcat crash

### ISSUE THREAD

**Issue by venkateshyella** [spring-projects/spring-framework#23716]
Hi,
I am having issues with tomcat crash every time when load occurs on JVM, I am getting below error logs initially it is working fine. When some users are increased to access the server then server crashing every time. My java version is 8.0_222 please help me to get out of this issue

I am using spring 4.0.0 version 

```
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f5e79885cb8, pid=3364, tid=0x00007f5df6fef700
#
# JRE version: OpenJDK Runtime Environment (8.0_222-b10) (build 1.8.0_222-b10)
# Java VM: OpenJDK 64-Bit Server VM (25.222-b10 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [core.so+0x7cb8]  socket_send+0x98
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /opt/apache-tomcat-8.0.33/bin/hs_err_pid3364.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
Sep 23, 2019 11:55:39 -0700 [5979 1] com.newrelic INFO: New Relic Agent: Loading configuration file "/opt/apache-tomcat-8.0.33/newrelic/./newrelic.yml"
Sep 23, 2019 11:55:40 -0700 [5979 1] com.newrelic INFO: New Relic Agent: Writing to log file: /opt/apache-tomcat-8.0.33/newrelic/logs/newrelic_agent.log
[GC (Allocation Failure) [ParNew: 545344K->35317K(613440K), 0.0337966 secs] 545344K->35317K(5174784K), 0.0338674 secs] [Times: user=0.12 sys=0.01, real=0.04 secs] 
[GC (CMS Initial Mark) [1 CMS-initial-mark: 0K(4561344K)] 35327K(5174784K), 0.0046687 secs] [Times: user=0.01 sys=0.00, real=0.00 secs] 
[CMS-concurrent-mark-start]
[CMS-concurrent-mark: 0.001/0.001 secs] [Times: user=0.01 sys=0.00, real=0.00 secs] 
[CMS-concurrent-preclean-start]
[CMS-concurrent-preclean: 0.005/0.005 secs] [Times: user=0.03 sys=0.00, real=0.01 secs] 
[CMS-concurrent-abortable-preclean-start]
23-Sep-2019 11:55:44.925 WARNING [main] org.apache.catalina.startup.SetAllPropertiesRule.begin [SetAllPropertiesRule]{Server/Service/Connector} Setting property 'keepAlivetimeout' to '60000' did not find a matching property.
23-Sep-2019 11:55:44.940 WARNING [main] org.apache.catalina.startup.SetAllPropertiesRule.begin [SetAllPropertiesRule]{Server/Service/Connector} Setting property 'keepAlivetimeout' to '60000' did not find a matching property.
23-Sep-2019 11:55:45.033 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version:        Apache Tomcat/8.0.33
23-Sep-2019 11:55:45.033 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server built:          Mar 18 2016 20:31:49 UTC
23-Sep-2019 11:55:45.033 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server number:         8.0.33.0
23-Sep-2019 11:55:45.033 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Name:               Linux
23-Sep-2019 11:55:45.033 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Version:            3.10.0-957.27.2.el7.x86_64
23-Sep-2019 11:55:45.033 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Architecture:          amd64
23-Sep-2019 11:55:45.033 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Java Home:             /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.222.b10-0.el7_6.x86_64/jre
23-Sep-2019 11:55:45.034 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Version:           1.8.0_222-b10
23-Sep-2019 11:55:45.034 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Vendor:            Oracle Corporation
23-Sep-2019 11:55:45.034 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_BASE:         /opt/apache-tomcat-8.0.33
23-Sep-2019 11:55:45.034 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_HOME:         /opt/apache-tomcat-8.0.33
```

**Comment by rstoyanchev** [comment]
I've edited your comment to improve the formatting. You might want to check out this [Mastering Markdown guide](https://guides.github.com/features/mastering-markdown/) for future reference.

**Comment by rstoyanchev** [comment]
@venkateshyella , first of all this is the issue tracker for the Spring Framework project. Spring Security is a separate project with its own issue tracker. What makes you believe the issue is related to Spring Security or to the Spring Framework? There is nothing Spring related in the log.

**Comment by venkateshyella** [comment]
Hi @rstoyanchev ,

Yes you are correct the error message does not have any spring related stuff but once I integrate spring security, then onwards this issue comes

**Comment by rstoyanchev** [comment]
Fine but without any description or sample, and only a snippet of a JVM crash log without anything Spring related, there isn't anything I can go on. You'll need do to more to narrow the problem down or give us something we can work with.

**Comment by venkateshyella** [comment]
@rstoyanchev 
I am using spring 4.0.0 RELEASE version and security also the same version.

It is working fine in local tomcat servers but Prod also 1 day working fine then real-time users are started using then this issue repeatedly occurring.

**Comment by venkateshyella** [comment]
Please find my security XML

<?xml version="1.0" encoding="UTF-8"?>
<beans:beans
	xmlns="http://www.springframework.org/schema/security"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:beans="http://www.springframework.org/schema/beans"
	xmlns:sec="http://www.springframework.org/schema/security"
	xmlns:aop="http://www.springframework.org/schema/aop"
	xsi:schemaLocation="
    http://www.springframework.org/schema/security
    http://www.springframework.org/schema/security/spring-security.xsd
    http://www.springframework.org/schema/beans
    http://www.springframework.org/schema/beans/spring-beans.xsd
    http://www.springframework.org/schema/aop 
	http://www.springframework.org/schema/aop/spring-aop.xsd">
	<sec:http pattern="/loginController/**" security="none"></sec:http>
	<sec:http pattern="/login/**" security="none"></sec:http>
	<sec:http pattern="/auth/**" security="none"></sec:http>
	<sec:http pattern="https://ok1static.oktacdn.com/assets/js/sdk/okta-auth-js/2.0.1/okta-auth-js.min.js" security="none"></sec:http>
	<sec:http pattern="/index" security="none"></sec:http>
	<sec:http pattern="/ext" security="none"></sec:http>
	<sec:http pattern="/resources/**" security="none"></sec:http>
	<sec:global-method-security proxy-target-class="true"  pre-post-annotations="enabled" secured-annotations="enabled"  jsr250-annotations="enabled"/>
	<sec:http use-expressions="true" auto-config="true" >
		    <sec:intercept-url pattern="/js/**" access="permitAll" />
		    <sec:intercept-url pattern="/css/**" access="permitAll" />
		    <sec:intercept-url pattern="/build/**" access="permitAll" />
		    <sec:intercept-url pattern="/ext/**" access="permitAll" />
			<sec:intercept-url pattern="/sass/**" access="permitAll" />
			<sec:intercept-url pattern="/modern/**" access="permitAll" />
			<sec:intercept-url pattern="/classic/**" access="permitAll" />
			<sec:intercept-url pattern="/META-INF/**" access="permitAll" />
			<sec:intercept-url pattern="/overrides/**" access="permitAll" />
			<sec:intercept-url pattern="/packages/**" access="permitAll" />
			<sec:intercept-url pattern="/resources/**" access="permitAll" />
			<sec:intercept-url pattern="/app/**" access="permitAll" />
			<sec:intercept-url pattern="/*.js" access="permitAll" />
			<sec:intercept-url pattern="/*.css" access="permitAll" />
			<sec:intercept-url pattern="/*.ico" access="permitAll" />
			<sec:intercept-url pattern="/*.json" access="permitAll" />

			<sec:intercept-url pattern="/**" access="hasAnyRole('ROLE_USER', 'ROLE_ADMIN')" />
			<sec:csrf token-repository-ref="csrfTokenRepository" disabled="true" />
			<sec:form-login login-page="/login" authentication-failure-url="/error"/>
			
			<sec:http-basic entry-point-ref="myBasicAuthenticationEntryPoint" />
	</sec:http>
	

	
	
	<beans:bean id="myUserDetailsService"
		class="com.capspay.config.MyUserDetailsService">
	</beans:bean>

	<authentication-manager
		alias="authenticationManager">
		<authentication-provider
			user-service-ref="myUserDetailsService">
			<password-encoder ref="passwordEncoder">
			</password-encoder>
		</authentication-provider>
	</authentication-manager>

	<beans:bean id="passwordEncoder"
		class="com.capspay.config.CustomPasswordEncoder">
		<beans:constructor-arg value="11" />
	</beans:bean>
	
	<beans:bean id="csrfTokenRepository"
	class="org.springframework.security.web.csrf.HttpSessionCsrfTokenRepository">
	<beans:property name="headerName" value="_csrf" />
</beans:bean>

</beans:beans>

**Comment by rstoyanchev** [comment]
@venkateshyella please refrain from commenting further here until you have narrowed things down more and have more concrete information.

**Comment by venkateshyella** [comment]
> @venkateshyella please refrain from commenting further here until you have narrowed things down more and have more concrete information.

Thanks got it I will post in relative tracker
